maestro-client/src/main/java/maestro/drivers/AndroidDriver.kt

```diff
@@ -522,7 +522,25 @@ class AndroidDriver(
         return false
     }
 
-    override fun waitForAppToSettle(initialHierarchy: ViewHierarchy?): ViewHierarchy? {
+    override fun waitForAppToSettle(initialHierarchy: ViewHierarchy?, uiElement: UiElement?, appId: String?): ViewHierarchy? {
+        return if (appId != null) {
+            waitForWindowToSettle(appId, initialHierarchy)
+        } else {
+            ScreenshotUtils.waitForAppToSettle(initialHierarchy, this)
+        }
+    }
+
+    private fun waitForWindowToSettle(appId: String, initialHierarchy: ViewHierarchy?): ViewHierarchy {
+        val endTime = System.currentTimeMillis() + WINDOW_UPDATE_TIMEOUT_MS
+
+        do {
+            if (blockingStub.isWindowUpdating(checkWindowUpdatingRequest { this.appId = appId }).isWindowUpdating) {
+                ScreenshotUtils.waitForAppToSettle(initialHierarchy, this)
+            } else {
+                return ViewHierarchy.from(this)
+            }
+        } while (System.currentTimeMillis() < endTime)
```

Comment:
Implementation of `MaestroTimer.withTimeout` is slightly different. It returns immediately in the if block but we want to wait until the screen is settled

Reply:
```
do {
   if (blockingStub.isWindowUpdating(checkWindowUpdatingRequest { this.appId = appId }).isWindowUpdating) {
       ScreenshotUtils.waitForAppToSettle(
           initialHierarchy, 
           this
       ) --> MaestroTimer.withTimeout returns here immediately
   } else {
       return ViewHierarchy.from(this)
   }
} while (System.currentTimeMillis() < endTime)
```

Reply:
`MaestroTimer` will return because your block returns a non-null value. You could just tweak it to return a `null` value if `isWindowUpdating` is still true after `waitForAppToSettle`

Reply:
My intent was to actually to `ScreenshotUtils.waitForAppToSettle` until we get true for isWindowUpdating because that method ensures we stabilize soon by comparing view hierarchies and screenshots